code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -1025,6 +1032,20 @@ pub mod pallet {
 					// position, that way any rounding is accounted for.
 					let new_stake = left_from_one_ratio.mul_ceil(existing_position.stake);
 					let new_share = left_from_one_ratio.mul_ceil(existing_position.share);
+
+					let rewards_pool = RewardPools::<T>::get(existing_position.reward_pool_id)
+						.ok_or(Error::<T>::RewardsPoolNotFound)?;
+
+					ensure!(
+						ratio.mul_floor(existing_position.stake) >=
+							rewards_pool.minimum_staking_amount,
+						Error::<T>::StakedAmountTooLow
+					);
+					ensure!(
+						new_stake >= rewards_pool.minimum_staking_amount,
+						Error::<T>::StakedAmountTooLow
+					);
```

Comment:
We could consider a different error here perhaps - something like `StakedAmountTooLowAfterSplit`?

Reply:
We could also add it to the docs on `StakedAmountTooLow` that it applies to both staking and splitting

Reply:
Whatever option you prefer it's fine for me, let me know and I change it

Reply:
For now I'll go with the first one, as a more specific seems always better

Reply:
I prefer the more specific error personally, so that works for me :+1: